kolena/_utils/krequests.py

```diff
@@ -52,11 +52,17 @@
 # Using the Retry object to configure a backoff which is not supported by using an int here.
 MAX_RETRIES = Retry(total=3, connect=3, read=0, redirect=0, status=0, backoff_factor=2)
 
+class NoOpAuth(requests.auth.AuthBase):
+    """Force requests to bypass `netrc` auth"""
+    def __call__(self, r):
+        return r
```

Comment:
Since we're setting the `"Authorization"` header below, rather than implementing this as a passthrough, should this actually set our Kolena bearer token? This seems to be the intended pattern (see [`auth.py` from psf/requests](https://github.com/psf/requests/blob/839a8edec37c81a18ac8332cfbd44f44e1ae6206/src/requests/auth.py#L95C28-L95C28)):

```py
class HTTPBasicAuth(AuthBase):
    """Attaches HTTP Basic Authentication to the given Request object."""

    def __init__(self, username, password):
        self.username = username
        self.password = password

    ...

    def __call__(self, r):
        r.headers["Authorization"] = _basic_auth_str(self.username, self.password)
        return r
```

Reply:
Completely agree. Will update